src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/listeners/PrisonerDomainEventsListener.kt

```diff
@@ -34,67 +35,43 @@ class PrisonerDomainEventsListener(
-        val (eventType, prisonerId) = objectMapper.readValue<HMPPSDomainEvent>(sqsMessage.Message)
+        val (eventType, prisonerId, incentiveId) = objectMapper.readValue<HMPPSDomainEvent>(sqsMessage.Message)
         telemetryClient.trackEvent(
           "prisoner-domain-event-received",
-          mapOf("eventType" to eventType, "offenderNo" to prisonerId),
+          mapOf(
+            "eventType" to eventType,
+            "offenderNo" to (prisonerId ?: ""),
+            "id" to (incentiveId?.toString() ?: "")
+          ),
         )
         if (eventFeatureSwitch.isEnabled(eventType)) when (eventType) {
           "prison-visit.booked" -> prisonVisitsService.createVisit(objectMapper.readValue(sqsMessage.Message))
           "prison-visit.cancelled" -> prisonVisitsService.cancelVisit(objectMapper.readValue(sqsMessage.Message))
+          "incentive.created" -> incentivesService.createIncentive(objectMapper.readValue(sqsMessage.Message))
           else -> log.info("Received a message I wasn't expecting {}", eventType)
         } else {
           log.warn("Feature switch is disabled for {}", eventType)
         }
       }
 
       "RETRY" -> {
-        val context = objectMapper.readValue<VisitContext>(sqsMessage.Message)
+        val context = objectMapper.readValue<Context>(sqsMessage.Message)
         telemetryClient.trackEvent(
           "prisoner-retry-received",
-          mapOf("vsipId" to context.vsipId, "nomisId" to context.nomisId),
+          mapOf("type" to context.type, "id" to context.getId()),
         )
-        prisonVisitsService.createVisitRetry(context)
-      }
-
-      "Incentive" -> {
-        val (eventType, id) = objectMapper.readValue<HMPPSIncentiveDomainEvent>(sqsMessage.Message)
-        telemetryClient.trackEvent(
-          "incentive-domain-event-received",
-          mapOf("eventType" to eventType, "id" to id),
-        )
-        if (eventFeatureSwitch.isEnabled(eventType)) when (eventType) {
-          "incentive.created" -> incentivesService.createIncentive(objectMapper.readValue(sqsMessage.Message))
-          else -> log.info("Received a message I wasn't expecting {}", eventType)
-        } else {
-          log.warn("Feature switch is disabled for {}", eventType)
+        when (context.type) {
+          "VISIT" -> prisonVisitsService.createVisitRetry(context as VisitContext)
+          "INCENTIVE" -> incentivesService.createIncentiveRetry(context as IncentiveContext)
         }
       }
-
-      "IncentiveRETRY" -> {
-        val context = objectMapper.readValue<IncentiveContext>(sqsMessage.Message)
-        telemetryClient.trackEvent(
-          "incentive-retry-received",
-          mapOf(
-            "id" to context.incentiveId.toString(),
-            "nomisBookingId" to context.nomisBookingId.toString(),
-            "nomisIncentiveSequence" to context.nomisIncentiveSequence.toString()
-          ),
-        )
-        incentivesService.createIncentiveRetry(context)
-      }
```

Comment:
rearranged these when I realised both visit nd incentive messages will have the same 'type' but different 'eventType'